src/popup/components/Main.vue

```diff
@@ -217,368 +213,401 @@
-    async getUserOptions() {
-      this.loading = true;
+// get alias options and mailboxes
+const getUserOptions = async () => {
+  loading.value = true;
 
-      const results = await Promise.all([
-        callAPI(
-          API_ROUTE.GET_ALIAS_OPTIONS,
-          {
-            hostname: this.hostName,
-          },
-          API_ON_ERR.TOAST
-        ),
-        callAPI(API_ROUTE.GET_MAILBOXES, {}, API_ON_ERR.TOAST),
-      ]);
+  const results = await Promise.all([
+    callAPI(
+      API_ROUTE.GET_ALIAS_OPTIONS,
+      {
+        hostname: hostName.value,
+      },
+      API_ON_ERR.TOAST
+    ),
+    callAPI(API_ROUTE.GET_MAILBOXES, {}, API_ON_ERR.TOAST),
+  ]);
 
-      const aliasOptions = results[0].data;
-      const { mailboxes } = results[1].data;
+  const aliasOptions = results[0].data;
+  const mailboxesData = results[1].data;
 
-      if (aliasOptions.recommendation) {
-        this.recommendation.show = true;
-        this.recommendation.alias = aliasOptions.recommendation.alias;
-      }
+  if (aliasOptions.recommendation) {
+    recommendation.show = true;
+    recommendation.alias = aliasOptions.recommendation.alias;
+  }
 
-      this.aliasSuffixes = aliasOptions.suffixes;
-      this.signedSuffix = this.aliasSuffixes[0];
-      this.aliasPrefix = aliasOptions.prefix_suggestion;
-      this.canCreate = aliasOptions.can_create;
-      this.mailboxes = mailboxes;
+  aliasSuffixes.value = aliasOptions.suffixes;
+  signedSuffix.value = aliasSuffixes.value[0];
+  aliasPrefix.value = aliasOptions.prefix_suggestion;
+  canCreate.value = aliasOptions.can_create;
+  mailboxes.value = mailboxesData;
 
-      this.loading = false;
+  loading.value = false;
 
-      await this.loadAlias();
+  await loadAlias();
 
-      tippy(".recommended-alias", {
-        content: "Click to copy",
-        placement: "bottom",
-      });
-    },
+  tippy(".recommended-alias", {
+    content: "Click to copy",
+    placement: "bottom",
+  });
+}
+
+    /**
+ * merge newAliases into currentAliases. If conflict, keep the new one
+ * 
+ * @param currentAliases 
+ * @param newAliases 
+ */
+const mergeAliases = (currentAliases, newAliases) => {
+  // dict of aliasId and alias to speed up research
+  let newAliasesDict = {};
+  for (var i = 0; i < newAliases.length; i++) {
+    let alias = newAliases[i];
+    newAliasesDict[alias.id] = alias;
+  }
 
-    async loadAlias() {
-      const contentElem = this.contentElem;
-      this.aliasArray = [];
-      let currentPage = 0;
+  let ret = [];
 
-      this.aliasArray = await this.fetchAlias(currentPage, this.searchString);
+  // keep track of added aliases
+  let alreadyAddedId = {};
+  for (var i = 0; i < currentAliases.length; i++) {
+    let alias = currentAliases[i];
+    if (newAliasesDict[alias.id]) ret.push(newAliasesDict[alias.id]);
+    else ret.push(alias);
 
-      let allAliasesAreLoaded = false;
+    alreadyAddedId[alias.id] = true;
+  }
 
-      let that = this;
```

Comment:
Oh yeah! Also, check out what this is, I'm not sure what the point of declaring this is. I thought this may have been to break reactivity in this function? But then I thought, what would be the point in that? I didn't do that in my version of the code -- there is no "this" context in script setup, so doing that is a bit difficult, and I didn't really understand the purpose.